scripts/bin/auto-mount

```diff
@@ -1,17 +1,52 @@
-#! /bin/sh
-
-sleep 5
-
-$SNAP/usr/local/bin/udisksctl monitor | grep --line-buffered " Device:" | while read -r -- device; do
-
-  echo "$SNAP_NAME.auto-mount found ${device}"
-  if echo $device| grep -q "[0-9]$"; then
-    DEV="$(echo $device | sed 's/^Device: //')"
-  fi
-  if [ -n "$DEV" ]; then
-    RES=`$SNAP/usr/local/bin/udisksctl mount --no-user-interaction --block-device "$DEV"`
-    # process returned value into the mount point string
-    RES2=`echo $RES | cut -f4 -d\  | cut -f1 -d.`
-    echo $RES2 > $WATCHED
-  fi
-done
+#!/bin/sh
+
+. "$SNAP/bin/common"
+
+init_mount_dir() {
+    _arg_dir="$1"
+
+}
+
+watch_devices() {
+    inotifywait --quiet --monitor --event "moved_to" --event "delete" /dev/disk/by-uuid | \
+        while read -r _parent _action _uuid; do
+            case "$_action" in
+                MOVED_TO) try_mount "$_parent" "$_uuid" ;;
+                DELETE)   rm_mount  "$_uuid" ;;
```

Comment:
preemptive `rm_mount not found`

Reply:
Note from conversation: unmounting is hard to do in a clean fashion (no good way of indicating a "finished" state). What we can instead do it is unmount the partition and trigger a system reboot, which is a fine enough indication that the process has finished.

We can leave a comment indicating other options one can take, but we shouldn't implement them.

Reply:
We should think about what to do in cases where no snaps are imported. Unfortunately, the best option is probably to simply avoid rebooting in that case, otherwise we get reboots even if we're just trying to import user assertions.

Reply:
Also, this could potentially be a DoS attack, if someone just leaves in a USB drive 😂